Scripts/Process/Component/GameObjectSourceTargetProcessor.cs

```diff
@@ -0,0 +1,187 @@
+﻿namespace VRTK.Core.Process.Component
+{
+    using UnityEngine;
+    using System.Collections.Generic;
+    using VRTK.Core.Extension;
+    using VRTK.Core.Rule;
+
+    public abstract class GameObjectSourceTargetProcessor : SourceTargetProcessor<GameObject, GameObject>
+    {
+        #region Processor Settings
+        /// <summary>
+        /// A <see cref="GameObject"/> collection of sources to apply data from.
+        /// </summary>
+        [Header("Entity Settings"), Tooltip("A GameObject collection of sources to apply data from.")]
+        public List<GameObject> sources = new List<GameObject>();
+        /// <summary>
+        /// Allows to optionally determine which sources should be processed based on the set rules.
+        /// </summary>
+        [Tooltip("Allows to optionally determine which sources should be processed based on the set rules.")]
+        public RuleContainer sourceValidity;
+        /// <summary>
+        /// A <see cref="GameObject"/> collection of targets to apply data to.
+        /// </summary>
+        [Tooltip("A GameObject collection of targets to apply data to.")]
+        public List<GameObject> targets = new List<GameObject>();
+        /// <summary>
+        /// Allows to optionally determine which targets should be processed based on the set rules.
+        /// </summary>
+        [Tooltip("Allows to optionally determine which targets should be processed based on the set rules.")]
+        public RuleContainer targetValidity;
+        #endregion
+
+        /// <summary>
+        /// The current sources collection index.
+        /// </summary>
+        public int CurrentSourcesIndex
+        {
+            get { return _currentSourcesIndex; }
+            set { _currentSourcesIndex = sources.GetWrappedAndClampedIndex(value); }
+        }
+        private int _currentSourcesIndex;
+
+        /// <summary>
+        /// The current targets collection index.
+        /// </summary>
+        public int CurrentTargetsIndex
+        {
+            get { return _currentTargetsIndex; }
+            set { _currentTargetsIndex = targets.GetWrappedAndClampedIndex(value); }
+        }
+        private int _currentTargetsIndex;
+
+        /// <inheritdoc />
+        public override void Process()
```

Comment:
Does it make sense to move this into the superclass instead? Subclasses can still override it, but it seems like a good default implementation of this method to have in `SourceTargetProcessor`.

Reply:
Only the `!isActiveAndEnabled` because the `sources` and `target` do not exist in the superclass.

Reply:
But then doing the return in the superclass won't help because it will just exit early from the superclass method and keep running this method anyway.

Reply:
👍